lib/screens/categories/categories_viewmodel.dart

```diff
@@ -0,0 +1,19 @@
+import 'package:boilerplate/screens/categories/states/categories_state.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:boilerplate/services/cocktail_service.dart';
+
+class CategoriesViewModel extends StateNotifier<CategoriesState> {
+  final CocktailServiceInterface cocktailService;
+
+  CategoriesViewModel(super.state, {required this.cocktailService});
```

Comment:
Nice! I see you are using `super.state` and that came new on `Flutter 3`!! That's sweet! 🚀